### PR TITLE
Add E2E test

### PR DESCRIPTION
**Pull request by jhen0409**
1. Add E2E test, using `chromedriver`, `selenium-webdriver`
 - Electron v0.33.7+ support `--app` option, so we can use `electron --app=.`  to start main process in `chromedriver`.
2. Unit tests command add `$(find test -name '*.spec.js')`, because the original will run e2e test file. (I don't recommend using E2E test in watch mode)
3. Add generateScopedName to `./test/setup.js` (css-modules-require-hook), because E2E test will find elements with className.

#### Usage
```bash
// start chromedriver
npm run test-e2e:crdv

// new console tab
npm run build && npm run test-e2e
```

If we want it to be test in the Travis, can add following code to travis file:
```yml
...
install:
...
  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16"

before_script:
...
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start &
  - sleep 3
  - npm run test-e2e:crdv &
  - sleep 3

script:
...
  - npm run test-e2e
```

### CODE REVIEW

**Comment by chentsulin** [comment]
Awesome works! I agree to adding e2e test to travis, could you add a commit for it?

File: package.json [hunk 1]
```diff
@@ -2,10 +2,12 @@
   "name": "electron-react-boilerplate",
   "version": "0.7.0-beta",
   "description": "Electron application boilerplate based on React, React Router, Webpack, React Hot Loader for rapid application development",
-  "main": "main.js",
+  "main": "main.js",
```
Comment (chentsulin) [inline package.json]: This space is unnecessary.

**Comment by jhen0409** [comment]
@chentsulin it's done. :)

**Comment by chentsulin** [comment]
Great! Thank you :+1:

**Comment by chentsulin** [comment]
Oops, I have no ideas why test job failed on the master branch. @jhen0409 Could you try to fix it? I already add you as collaborator

**Comment by jhen0409** [comment]
Oh... [this job](https://travis-ci.org/chentsulin/electron-react-boilerplate/jobs/90509552) seem to be keyserver problem.

```
$ sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --secret-keyring /tmp/tmp.7BaGE7eQxP --trustdb-name /etc/apt/trustdb.gpg --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d//apt.postgresql.org.gpg --keyserver hkp://keyserver.ubuntu.com:80/ --recv 60C317803A41BA51845E371A1E9377A2BA9EF27F
gpg: requesting key BA9EF27F from hkp server keyserver.ubuntu.com
gpgkeys: key 60C317803A41BA51845E371A1E9377A2BA9EF27F not found on keyserver
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
```

@chentsulin you can try restart job, it should be no longer error. :)

**Comment by chentsulin** [comment]
It's passed now. :+1:

**Pull request merged**